### PR TITLE
Fix prettier detection in yarn workspaces with PnP

### LINKED ISSUE #46: Can't find prettier with PnP in workspace child
With the following setup:

```
workspace/package.json          # Workspace root with `"workspaces": ["child"]`
workspace/.pnp.js               # Dependency modules live at workspace root
workspace/child/package.json    # Workspace child with `"dependencies": { "prettier": "..." }`
```

`getYarnPnpifyedLocalPrettier` fails to find prettier because when it runs with directory `workspace/child` [the checks for `.pnp.js`](https://github.com/jscheid/prettier.el/blob/4c285867ea4c293eca34b5714e93d15f715211ad/prettier-el.js#L203-L207) fail, because the `.pnp.js` file lives in the workspace root. It then tries the parent directory (workspace root), but there's no `prettier` dependency at that level, causing the following error:

```Error: Your application tried to access prettier, but it isn't declared in your dependencies; this makes the require call ambiguous and unsound.```

@asbish since you wrote the PnP code, do you have any thoughts on how to best fix this? One idea is to check for `package.json` instead of `.pnp.js` and if found, run `yarn config get pnpMode` to decide whether PnP is enabled, wdyt?

### PR DESCRIPTION
**Pull request by jscheid**
Closes #46

@asbish I think this should work, but need to test it more with older Node versions etc.

Curious to hear what you think of it.

### CODE REVIEW

**Comment by asbish** [comment]
Thank you very much @jscheid, I totally agree.
I've checked below three cases in Node v12.19.0 and these work correctly under this commit.
1.
```
workspace/package.json # `"dependencies": {"prettier: "2.1.2"}, "workspaces": ["child", "child1"]`
workspace/child/package.json # `"dependencies": {}`
workspace/child1/package.json # `"dependencies": {}`
```
2.
```
workspace/package.json # `"dependencies": {}, "workspaces": ["child", "child1"]`
workspace/child/package.json # `"dependencies": {"prettier: "2.1.2"}`
workspace/child1/package.json # `"dependencies": {}`
```
3.
```
workspace/package.json # `"dependencies": {}, "workspaces": ["child", "child1"]`
workspace/child/package.json # `"dependencies": {"prettier: "2.1.2"}`
workspace/child1/package.json # `"dependencies": {"prettier: "2.1.0"}`
```
I didn't expect there was a possibility that a project would have several `prettier` versions then. One of the reasons was `yarn pnpify --sdk` command only supports root `package.json`.
By the way, if `.yarn/sdks/prettier` the command generates is changed, we'll need to add some fix to `prettier.el`.

File: prettier-el.js [hunk 1]
```diff
@@ -174,54 +183,83 @@ function getGlobalPrettier() {
...
-      if (!process["versions"]["pnp"]) {
-        externalRequire(yarnPnpCandidate)["setup"]();
-      }
-      const targetRequire = createRequire(yarnPnpCandidate);
-      return targetRequire(targetRequire["resolve"]("prettier"));
-    } catch (e) {
-      if (e.code === "ENOENT") return null;
-      throw e;
-    }
+function getLocalPrettier(directory) {
```
Comment (asbish) [inline prettier-el.js:237]: There are trailing white spaces.

**Review by asbish — APPROVED** [review]

File: prettier-el.js (hunk 1 above)
Comment (jscheid) [reply prettier-el.js:237]: Do you mean the double space before `The`? I did that on purpose, to separate the description from the argument name. Or what do you mean?

Comment (asbish) [reply prettier-el.js:237]: I'm sorry for not explaining enough.
These spaces before `{`
```
function getLocalPrettier(directory) {<space><space>
```

Comment (jscheid) [reply prettier-el.js:237]: Oh, yes you're right. That's really weird. I'm using `whitespace-cleanup-mode` and I could have sworn that GitHub highlights trailing spaces in the diff view. So it seems that two completely unrelated tools let me down!

Fixing this now...

**Comment by jscheid** [comment]
@asbish thank you for the thoughtful review. I've been dogfooding this (via the branch tarball) and might do so for a couple days longer before I merge this, just to be extra sure it doesn't break anything.

>  By the way, if .yarn/sdks/prettier the command generates is changed, we'll need to add some fix to prettier.el.

I wasn't aware of the `yarn sdk` command until now, I'm going to take a closer look.

**Pull request merged**